### DISCUSSION TITLE
Filtering JSON fields in mongodb provider

### DISCUSSION THREAD

**Discussion by jzzfs** [prisma/prisma#8452] [Q&A] [answered]
Hi! MongoDB [JSON fields](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields) are already supported, but [JSON **filters**](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#json-filters) are not (unlike in psql and mysql).

So the only option left would be to execute a raw query -- either via `$queryRaw` or `executeRaw`. Yet, both of these throw:
```
Invalid `prisma.executeRaw()` invocation:
The current database provider doesn't support a feature that the query used: Raw queries 
```
So my questions are:
1. Is there any ETA for MongoDB JSON filters and/or raw query support?
2. If not, do I have to use another pkg (like [`mongoose`](https://www.npmjs.com/package/mongoose) or [`node-mongodb-native`](https://github.com/mongodb/node-mongodb-native/)) to account for prisma's limitations?

Thank you!

PS -- If it's of any help, I have both applicable `previewFeatures` turned on:
```
generator client {
    provider        = "prisma-client-js"
    previewFeatures = ["mongoDb", "filterJson"]
}

```

**Comment by janpio** [comment, marked as answer]
As MongoDB is in preview, there is some functionality missing - but of course we want to get it on par with other databases in Prisma as soon as possible.

- Raw query support in MongoDB is tracked here: https://github.com/prisma/prisma/issues/8270
- For JSON filters I could not quickly find an issue, so I suggest you open a new one with the required information.

**Reply by jzzfs** [reply]
Thank you, will do!

**Reply by reslear** [reply]
@jzzfs What did you use the solution?

**Reply by reslear** [reply]
https://github.com/prisma/prisma/issues/10879

**Reply by kylejw2** [reply]
Have any updates been made on this? The link you reference for raw query support is a dead end because you've said it's not possible.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
